pal/input/minigame/gamepad-input.ts

```diff
@@ -1,7 +1,124 @@
-export class GamepadInputSource {
-    support: boolean;
+import { GamepadCallback } from 'pal/input';
+import { InputEventType } from '../../../cocos/input/types/event-enum';
+import { EventTarget } from '../../../cocos/core/event/event-target';
+import { InputSourceButton, InputSourceDpad, InputSourceStick } from '../input-source';
 
-    constructor () {
-        this.support = false;
+export class GamepadInputDevice {
+    public static all: GamepadInputDevice[] = [];
+
+    public buttonNorth!: InputSourceButton;
```

Comment:
Public variable can be modified outside, is it any reason to use public variable?

Reply:
need to change to a  public getter